kotlin-generator/src/helpers.ts

```diff
@@ -0,0 +1,206 @@
+import { ArrayType, EnumType, OptionalType, StructType, Type, TypeReference } from "@sdkgen/parser";
+
+export function generateEnum(type: EnumType) {
+    return `enum class ${type.name} { ${type.values.map(x => mangle(x.value)).join(", ")} }\n`;
+}
+
+export function generateClass(type: StructType) {
+    return `data class ${type.name}(${type.fields.map(field => `val ${mangle(field.name)}: ${generateTypeName(field.type)}`).join(",  ")})\n`;
+}
+
+export function generateErrorClass(error: string) {
+    return `class ${error}(message: String): Error(message)\n`;
+}
+
+// Don`t know if will have use to this method yet.
+// export function cast(value: string, type: Type): string {
+//     if (type.constructor.name === "ArrayType") {
+//         return `(${value} as List).map((e) => ${cast("e", (type as ArrayType).base)}).toList()`;
+//     } else if (type.constructor.name === "VoidPrimitiveType") {
+//         return value;
+//     } else if (type.constructor.name === "FloatPrimitiveType" || type.constructor.name === "MoneyPrimitiveType") {
+//         return `(${value} as num)?.toDouble()`;
+//     } else {
+//         return `${value} as ${generateTypeName(type)}`;
+//     }
+// }
+
+export function generateTypeName(type: Type): string {
+    switch (type.constructor.name) {
+        case "StringPrimitiveType":
+        case "CpfPrimitiveType":
+        case "CnpjPrimitiveType":
+        case "EmailPrimitiveType":
+        case "CepPrimitiveType":
+        case "UrlPrimitiveType":
+        case "UuidPrimitiveType":
+        case "HexPrimitiveType":
+        case "Base64PrimitiveType":
+        case "SafeHtmlPrimitiveType":
+        case "XmlPrimitiveType":
+            return "String";
+        case "IntPrimitiveType":
+        case "UIntPrimitiveType":
+        case "MoneyPrimitiveType":
+            return "Int";
+        case "FloatPrimitiveType":
+            return "Double";
+        case "DatePrimitiveType":
+        case "DateTimePrimitiveType":
+            return "Calendar";
+        case "BoolPrimitiveType":
+            return "Boolean";
+        case "BytesPrimitiveType":
+            return "ByteArray";
+        case "VoidPrimitiveType":
+            return "Unit";
+        case "AnyPrimitiveType":
+            return "Any";
```

Comment:
It should be JsonPrimitiveType. AnyPrimitiveType doesn't exist. Also, it would be better to return a JsonValue class of some kind (Gson has it?)

Reply:
Yes, I'm using it, it's called JsonObject. AnyPrimitiveType is being polled on dart target, this is why this is here 🤔 
Maybe we need to revise dart-generator to remove the unused and mismatches in type handling

Reply:
Your branch wasn't in sync with master, this is why you see them